lib/vms.js

```diff
@@ -299,20 +299,22 @@ function updateTags(req, res, next) {
     var params = {};
     params.uuid = req.params.uuid;
     params.metadata = req.body;
-
-    req.sdc[req.dc].vmapi.setMetadata(
-        'tags',
-        params,
-        { headers: getRequestHeaders(req) },
-        function (err, obj)
-    {
+    function vmapiCallback(err, obj) {
         if (err) {
             req.log.fatal(err);
             return next(err);
         }
         res.send(obj);
         return next();
-    });
+    }
+    if (!params.metadata) {
+        return req.sdc[req.dc].vmapi.deleteAllMetadata(
```

Comment:
This is not doing what is intended as the tag value is an empty json, not string. I deleted all tags and deleteAllMetadata didn't happen. Also setMetadata should be triggered in the else-condition. It got invoked and returned 409 for the empty json.